File: cpp/source/event/input/user/CloseChannels.cpp
#include "event/input/user/CloseChannels.hpp"

#include <sstream>
#include <iterator>

using namespace fm;
using namespace fm::event::input::user;

CloseChannels::CloseChannels() :
    UserEvent(CLOSE_CHANNELS)
{
}

CloseChannels::CloseChannels(const std::vector<long>& _channels) :
    UserEvent(CLOSE_CHANNELS),
    channels(_channels)
{
}

std::string CloseChannels::toString() const
{
    std::ostringstream oss;
    if (!channels.empty())
    {
        std::copy(channels.begin(), channels.end()-1,
                  std::ostream_iterator<long>(oss, ","));
        oss << channels.back();
    }
    return "CLOSE_CHANNELS: channels: " + oss.str();
}

std::vector<long>& CloseChannels::getChannels()
{
    return channels;
}

const std::vector<long>& CloseChannels::getChannels() const
{
    return channels;
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/input/user/CloseChannels.java
package com.fleetmgr.sdk.client.event.input.user;

import java.util.List;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class CloseChannels extends UserEvent {

    private List<Long> channels;

    public CloseChannels(List<Long> channels) {
        super(Type.CLOSE_CHANNELS);
        this.channels = channels;
    }

    public List<Long> getChannels() {
        return channels;
    }

    @Override
    public String toString() {
        return "CloseChannels{" +
                "channels=" + channels +
                '}';
    }
}
